### PR TITLE
add build_shared

### PR DESCRIPTION
**Pull request by matthiasdiener**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by matthiasdiener** [comment]
Is this what you had in mind @isuruf ?

**Comment by isuruf** [comment]
Yes. There's one change though. We need to rename the files.
```
%LIBRARY_LIB%\fltk.lib -> %LIBRARY_LIB%\fltk_static.lib
%LIBRARY_LIB%\fltk_SHARED.lib -> %LIBRARY_LIB%\fltk.lib
```

File: recipe/meta.yaml [hunk 1]
```diff
@@ -24,34 +24,31 @@ requirements:
...
     - {{ cdt('expat') }}             # [linux]
     - {{ cdt('mesa-dri-drivers') }}  # [linux]
     - freetype                       # [linux]
     - gnuconfig                      # [unix]
+    - xorg-libxext
+    - xorg-libxrender
+    - xorg-libxfixes
+    - libglu    # [linux]
+    - jpeg
+    - libpng
```
Comment (isuruf) [inline recipe/meta.yaml]: These should be host dependencies.

Comment (matthiasdiener) [reply recipe/meta.yaml]: Right, but for some reason putting them into `build:` fixes many of the linking issues when building FLTK with shared libraries. I'm not sure what is going on there. Maybe some kind of conflict between the X11-related  yum/cdt/conda packages?

File: recipe/meta.yaml [hunk 2]
```diff
@@ -73,6 +73,7 @@ requirements:
 test:
   commands:
     - test -f $PREFIX/lib/libfltk.a  # [unix]
+    - test -f $PREFIX/lib/libfltk.${SHLIB_EXT}  # [unix]
```
Comment (isuruf) [inline recipe/meta.yaml]:
```suggestion
    - test -f $PREFIX/lib/libfltk${SHLIB_EXT}  # [unix]
```

File: recipe/meta.yaml (hunk 1 above)
Comment (matthiasdiener) [reply recipe/meta.yaml]: Turns out fltk ignores LDFLAGS for shared libraries, it uses DSOFLAGS instead.

File: recipe/meta.yaml [hunk 3]
```diff
@@ -73,9 +73,11 @@ requirements:
 test:
   commands:
     - test -f $PREFIX/lib/libfltk.a  # [unix]
+    - test -f $PREFIX/lib/libfltk${SHLIB_EXT}  # [unix]
     - test -f $PREFIX/lib/libfltk_gl.a  # [unix]
-    - if not exist %LIBRARY_LIB%\\fltk.lib exit 1  # [win]
```
Comment (isuruf) [inline recipe/meta.yaml:77]: Why remove this?

Comment (matthiasdiener) [reply recipe/meta.yaml:77]: Isn't the static library file name changed here? https://github.com/conda-forge/fltk-feedstock/pull/12/files#diff-54cf74e113dd3f6d11e092fdb1d888ec82c69bdafbb15cfb6570c83ecad28f33R23-R26

File: recipe/bld.bat [hunk 4]
```diff
@@ -18,3 +19,9 @@ if errorlevel 1 exit 1
 :: Install.
 nmake install
 if errorlevel 1 exit 1
+
+move %LIBRARY_LIB%\fltk.lib %LIBRARY_LIB%\fltk_static.lib
+move %LIBRARY_LIB%\fltk_forms.lib %LIBRARY_LIB%\fltk_forms_static.lib
+move %LIBRARY_LIB%\fltk_images.lib %LIBRARY_LIB%\fltk_images_static.lib
+move %LIBRARY_LIB%\fltk_gl.lib %LIBRARY_LIB%\fltk_gl_static.lib
+
```
Comment (isuruf) [inline recipe/bld.bat]:
```suggestion
```

File: recipe/meta.yaml [hunk 5]
```diff
@@ -73,9 +73,11 @@ requirements:
...
-    - if not exist %LIBRARY_LIB%\\fltk_gl.lib exit 1  # [win]
+    - if not exist %LIBRARY_LIB%\\fltk_static.lib exit 1  # [win]
+    - if not exist %LIBRARY_LIB%\\fltk_gl_static.lib exit 1  # [win]
```
Comment (isuruf) [inline recipe/meta.yaml]:
```suggestion
    - if not exist %LIBRARY_LIB%\\fltk.lib exit 1  # [win]
    - if not exist %LIBRARY_LIB%\\fltk_gl.lib exit 1  # [win]
    - if not exist %LIBRARY_LIB%\\libfltk.lib exit 1  # [win]
```

Comment (matthiasdiener) [reply recipe/meta.yaml]: libfltk_gl.lib does not exist I think, this seems to be the full list:

```
-- Installing: D:/bld/fltk_1631216060244/_h_env/Library/lib/fltk.lib
-- Installing: D:/bld/fltk_1631216060244/_h_env/Library/lib/fltk_forms.lib
-- Installing: D:/bld/fltk_1631216060244/_h_env/Library/lib/fltk_images.lib
-- Installing: D:/bld/fltk_1631216060244/_h_env/Library/lib/fltk_gl.lib
-- Installing: D:/bld/fltk_1631216060244/_h_env/Library/lib/libfltk.lib
-- Installing: D:/bld/fltk_1631216060244/_h_env/Library/bin/libfltk.dll
```

Comment (isuruf) [reply recipe/meta.yaml]: Ah, okay. Can you check that those files exist? (It looks like only `libfltk` has a shared library`)

Comment (matthiasdiener) [reply recipe/meta.yaml]: Should I move bin/libfltk.dll to lib/libfltk.dll ?

Comment (isuruf) [reply recipe/meta.yaml]: Nope. `dll`s belong in `bin`

**Comment by matthiasdiener** [comment]
This is ready for review.

**Review by isuruf — APPROVED** [review]

**Pull request merged**